merino-wikipedia/src/es_helper.rs

```diff
@@ -0,0 +1,180 @@
+/// and more consistent.
+pub struct ElasticHelper {
+    /// The ES client this helper users.
+    pub client: Elasticsearch,
+    /// The search index this helper targets.
+    pub index_name: String,
+}
+
+impl ElasticHelper {
+    /// Create a new ElasticHelper that connects to the server specified in
+    /// `settings` and uses the index specified in [`index_name`].
+    ///
+    /// # Errors
+    /// If the settings to connect to Elasticsearch are not valid, the creation
+    /// process may fail.
+    pub fn new<S: Into<String>>(
+        es_settings: &merino_settings::ElasticsearchSettings,
+        index_name: S,
+    ) -> Result<Self> {
+        let transport_builder = match &es_settings.connection {
+            merino_settings::ElasticsearchConnection::Single { url } => {
+                TransportBuilder::new(SingleNodeConnectionPool::new(
+                    elasticsearch::http::Url::parse(&url.to_string())
+                        .context("Could not parse Elasticsearch URL")?,
+                ))
+            }
+            merino_settings::ElasticsearchConnection::Cloud { cloud_id } => TransportBuilder::new(
+                CloudConnectionPool::new(cloud_id)
+                    .context("Could not create Elasticsearch cloud connection")?,
+            ),
+            merino_settings::ElasticsearchConnection::None => {
+                bail!("Elasticsearch is not configured")
+            }
+        };
+
+        let transport = transport_builder
+            .build()
+            .context("misconfigured elasticsearch")?;
+
+        let client = Elasticsearch::new(transport);
+
+        Ok(Self {
+            client,
+            index_name: index_name.into(),
+        })
+    }
+
+    /// Check that the helper's index exists.
+    /// # Errors
+    /// If there is an HTTP error communication with ES.
+    pub async fn index_exists(&self) -> Result<bool> {
+        let exists_res = self
+            .client
+            .indices()
+            .exists(IndicesExistsParts::Index(&[&self.index_name]))
+            .send()
+            .await
+            .context(format!("index_exists({}) request", self.index_name))?;
+        match exists_res.status_code().as_u16() {
+            200 => Ok(true),
+            404 => Ok(false),
+            code => {
+                let exists_output = exists_res
+                    .text()
+                    .await
+                    .context("Fetching text of exists command")?;
+                Err(anyhow::anyhow!(
+                    "Unexpected status code from index_exists({}) {code}. Error: {exists_output}",
+                    self.index_name
+                ))
+            }
+        }
+    }
+
+    /// Creates the helper's index, assuming it doesn't already exist.
+    /// # Errors
+    /// If the index already exists, or if there is an HTTP error communication with ES.
+    pub async fn index_create(&self) -> Result<()> {
```

Comment:
I think it's fine to create the index here for now. We probably want to do it in a different place in the future for more flexibility.

Reply:
This is simply a helper method that codifies the creation of the index. We'll want to be intentional about where we call this long term of course, but it isn't actually linked to anything (except the demo) right now.